### PR TITLE
feat: Action redesign: Updating the config for SMTP plugin to use sections and zones format

### PR DESCRIPTION
**Pull request by ankitakinger**
## Description

Action redesign: Updating the config for SMTP plugin to use sections and zones format

Fixes [#35505](https://github.com/appsmithorg/appsmith/issues/35505)

## Automation

/ok-to-test tags="@tag.All"

### :mag: Cypress test results

> [!TIP]
> 🟢 🟢 🟢 All cypress tests have passed! 🎉 🎉 🎉
> Workflow run: <https://github.com/appsmithorg/appsmith/actions/runs/10659691143>
> Commit: 3917fdbff066acdcdf2104e0f690956a640254e5
> <a href="https://internal.appsmith.com/app/cypress-dashboard/rundetails-65890b3c81d7400d08fa9ee5?branch=master&workflowId=10659691143&attempt=2" target="_blank">Cypress dashboard</a>.
> Tags: `@tag.All`
> Spec:
> <hr>Mon, 02 Sep 2024 04:23:47 UTC

## Communication
Should the DevRel and Marketing teams inform users about this change?
- [ ] Yes
- [ ] No

## Summary by CodeRabbit

- **New Features**
	- Enhanced styling capabilities for the dynamic input text control with a new CSS class.
	- Redesigned SMTP plugin email configuration interface with a multi-section layout, improving usability and flexibility.
	- Added new fields for CC, BCC, and conditional visibility for reply settings in the email configuration.

- **Bug Fixes**
	- Removed minimum height and width constraints for dynamic input text elements, allowing for more flexible sizing.

### CODE REVIEW

File: app/client/src/pages/Editor/ActionForm/Zone/styles.module.css [hunk 1]
```diff
@@ -47,6 +47,11 @@
       }
     }
   }
+  /* DynamicInputTextControl min height and width removed */
+  & :global(.uqi-dynamic-input-text) {
```
Comment (alex-golovanov) [inline app/client/src/pages/Editor/ActionForm/Zone/styles.module.css:51]: Do you really need a global selector here?

Comment (ankitakinger) [reply app/client/src/pages/Editor/ActionForm/Zone/styles.module.css:51]: Yeah thats how it has been done for other fields... hence I have added it the similar way. This global lies under the selector `.zone`

**Comment by NilanshBansal** [comment]
Adding @AmanAgarwal041 for review

**Pull request closed without merging**